### PR TITLE
Remove kubeUtil references from logs agent launchers

### PR DESCRIPTION
**Pull request by juliogreff**
### What does this PR do?

We're trying to clean up some old code from the kubeUtil, so anything
that fetches pod information needs to be migrated to workloadmeta. This
commit removes all references to it from the docker and kubernetes
launchers, and removes a lot of retrying logic that's now redundant
since pods and containers are guaranteed to be present in the store by
the time the logs agent gets to know about them.

### Motivation

Cleaning up unnecessary calls to the kubelet now that workloadmeta is a viable (and better!) replacement.

### Describe how to test/QA your changes

Log collection still works. In kubernetes, both with container collect all enabled and disabled. In docker, with both container labels and pod annotations.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/logs/internal/launchers/docker/container.go [hunk 1]
```diff
@@ -10,21 +10,27 @@ package docker
...
 
 	"github.com/DataDog/datadog-agent/pkg/logs/config"
 	"github.com/DataDog/datadog-agent/pkg/logs/service"
 )
 
-// configPath refers to the configuration that can be passed over a docker label,
-// this feature is commonly named 'ad' or 'autodiscovery'.
-const configPath = "com.datadoghq.ad.logs"
+// configPath refers to the configuration that can be passed over a docker label or a pod annotation,
+// this feature is commonly named 'ad' or 'autodicovery'.
```
Comment (djmitche) [inline pkg/logs/internal/launchers/docker/container.go:28]: This should probably move after the `const (`

File: pkg/logs/internal/launchers/kubernetes/launcher.go [hunk 2]
```diff
@@ -242,43 +153,70 @@ func (l *Launcher) removeSource(service *service.Service) {
...
+	if annotation := l.getAnnotation(container.Name, pod.Annotations); annotation != "" {
 		configs, err := config.ParseJSON([]byte(annotation))
 		if err != nil || len(configs) == 0 {
 			return nil, fmt.Errorf("could not parse kubernetes annotation %v", annotation)
 		}
+
 		// We may have more than one log configuration in the annotation, ignore those
 		// unrelated to containers
-		containerType, _ := containers.SplitEntityName(container.ID)
+		containerType := string(runtimeContainer.Runtime)
```
Comment (djmitche) [inline pkg/logs/internal/launchers/kubernetes/launcher.go:190]: 😍

**Review by djmitche — APPROVED** [review]
I'm giddy to see that retry stuff removed from the k8s launcher!

Is dockerutil going away too?

**Review by gh123man — APPROVED** [review]

**Comment by juliogreff** [comment]
> Is dockerutil going away too?

kubeutil and dockerutil aren't going away per se, but ideally just getting some data about a container should go to workloadmeta instead. I think the refactoring you'll be working on will take care of it naturally for dockerutil in the logs agent, but if not I'll do another pass when it's all done to get rid of it too.

**Pull request merged**